File: src/main/cpp/commands/SetLeds.cpp
#include "commands/SetLeds.h"

/// @brief Command to set the LED mode.
/// @param Mode The LED mode.
/// @param m_leds The LED subsystem.
SetLeds::SetLeds(int Mode, Leds *leds) : m_mode(Mode), m_leds(leds)
{
    // Set the command name
    SetName("SetLeds");

    // Declare subsystem dependencies
    AddRequirements({m_leds});

}

/// @brief Command to set the LED mode.
/// @param Mode The LED mode.
/// @param m_leds The LED subsystem.
SetLeds::SetLeds(int Mode, units::second_t time, Leds *leds) : m_mode(Mode), m_time(time), m_leds(leds)
{
    // Set the command name
    SetName("SetLeds");

    // Declare subsystem dependencies
    AddRequirements({m_leds});

    // Remember the LED mode
    m_mode = Mode;

    // Indicate that the LED sequence has a time-out
    m_timed = true;
}

/// @brief Called just before this Command runs the first time.
void SetLeds::Initialize()
{
    // Set the LED mode
    m_leds->SetMode((LedMode) m_mode);

    // Get the LED sequence start time
    m_startTime = frc::GetTime();
}

/// @brief Indicates if the command has completed. Make this return true when this Command no longer needs to run execute().
/// @return True is the command has completed.
bool SetLeds::IsFinished()
{
    // Determine if a timed LED sequence
    if (m_timed == false)
        return false;

    // Determine if the LED sequence is complete
    if (frc::GetTime() - m_startTime > m_time)
        return true;

    return false;
}

/// @brief Called once after isFinished returns true.
/// @param interrupted Indicated that the command was interrupted.
void SetLeds::End(bool interrupted)
{
    
}

/// @brief Indicates if the command runs when the robot is disabled.
/// @return True is the command should run when the robot is disabled.
bool SetLeds::RunsWhenDisabled() const
{
    return true;
}

File: src/main/java/org/team3824/frc2025/commands/SetLeds.kt
package org.team3824.frc2025.commands

import edu.wpi.first.wpilibj2.command.Command

import org.team3824.frc2025.subsystems.LEDs
import org.team3824.frc2025.subsystems.LEDMode

/// @brief Command to set the LED mode.
/// @param Mode The LED mode.
/// @param m_leds The LED subsystem.
class SetLeds(private val m_mode: LEDMode,
              private val m_leds: LEDs): Command()
{

    /// @brief add requirements
    init
    {
        // Declare subsystem dependencies
        addRequirements(m_leds);
    }

    /// @brief Called just before this Command runs the first time.
    override fun initialize()
    {
        m_leds.SetMode(m_mode);
    }

    /// @brief Indicates if the command has completed. Make this return true when this Command no longer needs to run execute().
    /// @return True is the command has completed.
    override fun isFinished(): Boolean
    {
        return false;
    }

    /// @brief Called once after isFinished returns true.
    /// @param interrupted Indicated that the command was interrupted.
    override fun end(interrupted: Boolean)
    {
        m_leds.SetMode(LEDMode.Off)
    }

    /// @brief Indicates if the command runs when the robot is disabled.
    /// @return True is the command should run when the robot is disabled.
    override fun runsWhenDisabled(): Boolean
    {
        return true;
    }
}